next.config.mjs

```diff
@@ -24,6 +25,16 @@ const nextConfig = {
       child_process: false,
     };
 
+    if (!isServer) {
+      config.plugins.push(
+          new CopyPlugin({
+              patterns: [
+                  { from: 'app-config/production.yml', to: 'app-config/production.yml' },
```

Comment:
hardcoded file location (user states yaml name in env variable)

Reply:
This hardcoded file location is not used anymore, in [app/config/route.ts](https://github.com/biocypher/biochatter-next/blob/7a946b5c8e3957c3d49095841b86ad3d9a8e0b39/app/api/config/route.ts#L62), it will read env CUSTOM_BIOCHATTER_NEXT_FILE to get custom file location.

I'll remove this line later

Reply:
`docker compose build` fails for me due to that line:

```
 > [biochatter-next builder 5/5] RUN yarn build:
0.154 yarn run v1.22.19
0.171 $ cross-env BUILD_MODE=standalone next build
0.427 [Next] build mode standalone
0.428 [Next] build with chunk:  true
0.435 - warn You have enabled experimental feature (forceSwcTransforms) in next.config.mjs.
0.435 - warn Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
0.435
0.441 Attention: Next.js now collects completely anonymous telemetry regarding usage.
0.441 This information is used to shape Next.js' roadmap and prioritize features.
0.441 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
0.441 https://nextjs.org/telemetry
0.441
0.483 - info Creating an optimized production build...
1.691 Browserslist: caniuse-lite is outdated. Please run:
1.691   npx browserslist@latest --update-db
1.691   Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
2.449 Warning: For production Image Optimization with Next.js, the optional 'sharp' package is strongly recommended. Run 'yarn add sharp', and Next.js will use it automatically for Image Optimization.
2.449 Read more: https://nextjs.org/docs/messages/sharp-missing-in-production
24.49 Failed to compile.
24.49
24.49 unable to locate '/app/app-config/production.yml' glob
24.49
24.50
24.50 > Build failed because of webpack errors
24.55 error Command failed with exit code 1.
24.55 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
------
failed to solve: process "/bin/sh -c yarn build" did not complete successfully: exit code: 1
```

Reply:
be aware that I changed the location and file name of the config (and specified in `.env`)

Reply:
Let me remove the line

BTW, to make customization work in docker compose workflow, you don't need to modify .env.

To customize Biochatter-Next with this YAML file, we need to specify the location of our YAML file in docker-compose.yml:

...
services:
  biochatter-next:
    build:
      context: ./
      dockerfile: Dockerfile
    ports:
      - 3000:3000
    env_file:
      - .env
    volumes:
      - ./tmp:/tmp
      - ./app-config:/app-config            # mount the folder containing our YAML file (production.yml) to the container
    environment:
      - BASE_URL=http://flask-app:5001
      - CUSTOM_BIOCHATTER_NEXT_FILE=/app-config/production.yml  # tell BioChatter-Next where the YAML file is
    depends_on:
      - flask-app
 ...

Reply:
I am not sure I see the distinction. The contents of `.env` file (passed into the `env_file` field) should lead to the same result as the same variable(s) in the `environment` field, no? Should we just remove the `.env` file?

Reply:
I don't see this `- ./app-config:/app-config # mount the folder containing our YAML file (production.yml) to the container` in my `docker-compose.yaml`. Can you push all your changes to the repo?